addons/account_peppol/models/res_partner.py

```diff
@@ -0,0 +1,485 @@
+
+            if not partner.account_peppol_validity_last_check:
+                partner.account_peppol_verification_label = 'not_verified'
+            elif (
+                partner.ubl_cii_format
+                and participant_info is not None
+                and not partner._check_document_type_support(participant_info, partner.ubl_cii_format)
+            ):
+                # the partner might exist on the network, but not be able to receive that specific format
+                partner.account_peppol_verification_label = 'not_valid_format'
+            elif partner.account_peppol_is_endpoint_valid:
+                partner.account_peppol_verification_label = 'valid'
+            else:
+                partner.account_peppol_verification_label = 'not_valid'
+
+    # -------------------------------------------------------------------------
+    # BUSINESS ACTIONS
+    # -------------------------------------------------------------------------
+
+    @handle_demo
+    def button_account_peppol_check_partner_endpoint(self):
+        """ A basic check for whether a participant is reachable at the given
+        Peppol participant ID - peppol_eas:peppol_endpoint (ex: '9999:test')
+        The SML (Service Metadata Locator) assigns a DNS name to each peppol participant.
+        This DNS name resolves into the SMP (Service Metadata Publisher) of the participant.
+        The DNS address is of the following form:
+        - "http://B-" + hexstring(md5(lowercase(ID-VALUE))) + "." + ID-SCHEME + "." + SML-ZONE-NAME + "/" + url_encoded(ID-SCHEME + "::" + ID-VALUE)
+        (ref:https://peppol.helger.com/public/locale-en_US/menuitem-docs-doc-exchange)
+        """
+        self.ensure_one()
+
+        if not (self.peppol_eas and self.peppol_endpoint) or not self.ubl_cii_format:
+            self.account_peppol_is_endpoint_valid = False
+        else:
+            edi_identification = f'{self.peppol_eas}:{self.peppol_endpoint}'.lower()
+            self.account_peppol_validity_last_check = fields.Date.context_today(self)
+            self.account_peppol_is_endpoint_valid = bool(self._check_peppol_participant_exists(edi_identification, ubl_cii_format=self.ubl_cii_format))
+
+            if (
+                not self.account_peppol_is_endpoint_valid
+                and self.peppol_eas in ('0208', '9925')
+            ):
+                inverse_eas = '9925' if self.peppol_eas == '0208' else '0208'
+                inverse_endpoint = f'BE{self.peppol_endpoint}' if self.peppol_eas == '0208' else self.peppol_endpoint[2:]
+                if self._check_peppol_participant_exists(f'{inverse_eas}:{inverse_endpoint}', ubl_cii_format=self.ubl_cii_format):
+                    self.peppol_eas = inverse_eas
+                    self.peppol_endpoint = inverse_endpoint
+                    self.account_peppol_is_endpoint_valid = True
+        return False
+
+    # TODO: diff was a bit bad when copying the following fields in the right place above
```

Comment:
Fix the TODO and merge the two (`account_peppol`+ `account_edi_ubl_cii`) to respect guidelines of the file structure.